### PR TITLE
Collapse MLP layers in configuration files

### LINKED ISSUE #5: [Task2]: Improve the MLP constructor in VISSL by replacing 2mlp constructor with 1mlp constructor
**Task**: Currently in VISSL, if we want to stack a specific model layer named "mlp", we have to specify it multiple times in the config file. For example: https://github.com/facebookresearch/vissl/blob/master/configs/config/pretrain/simclr/models/regnet400mf.yaml#L15-L16 
However, this is not very intuitive a user and user expectation would be that following should just work.
```
HEAD:
      PARAMS: [
        ["mlp", {"dims": [440, 440, 128], "use_relu": True}],
      ]
```

Our goal in this task is to enable the above. 

**Steps/Tips**:
1. In the `mlp.py` file https://github.com/facebookresearch/vissl/blob/master/vissl/models/heads/mlp.py#L54-L55 , we need to replace the line 54 by `for i, dim in enumerate(dims[1:])` and  insert a check after line 55 `if i == (len(dims) - 2): break`. Make these changes in the `mlp.py`
2. Next, make changes in the config files under that show up in search https://github.com/search?q=%22mlp%22+repo%3Afacebookresearch%2Fvissl+path%3A%2Fconfigs%2Fconfig%2Fpretrain&type=Code&ref=advsearch&l=&l= to combine the "mlp" from 2 times to 1 time (as described in the task description above). Exception is that: don't make changes to the files in folders https://github.com/facebookresearch/vissl/tree/master/configs/config/pretrain/pirl/models and https://github.com/facebookresearch/vissl/tree/master/configs/config/pretrain/jigsaw/permutations
3. Run all the CPU unit tests and make sure they pass
4. Send a PR to VISSL and tag Priya as reviewer

### PR DESCRIPTION
**Pull request by akainth015**
Resolves #5 

```
Test Plan:

Command run:

Output:

```

### CODE REVIEW

File: tests/test_tasks.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 import pkg_resources
 from parameterized import parameterized
-from utils import UNIT_TEST_CONFIGS, SSLHydraConfig
+from .utils import UNIT_TEST_CONFIGS, SSLHydraConfig
```
Comment (prigoyal) [inline tests/test_tasks.py]: just curious, what's the motivation for this change?

**Review by prigoyal — CHANGES_REQUESTED** [review]
thank you for the PR @akainth015 and @grace-omotoso .  :) 

There is an issue with the current changes in `mlp.py`. As we discussed in the task https://github.com/MLH-Fellowship/vissl/issues/5, we have to add the check `if i == (len(dims) - 2): break` mandatory. If we don't add the check, then we introduce some unintended layers ReLU in the model. My suggestion is to follow something along the lines of what's proposed in the #5 .

Concretely, follow the 3 steps next:
1. Make changes to the `mlp.py` following the screenshot (NOTE: I introduced a small `skip_last_layer_relu_bn` so we can also handle the changes to the files that we skipped in #5 step 2.
Screen Shot 2021-02-27 at 1 53 46 PM

2. Make changes to the files in this folder https://github.com/facebookresearch/vissl/tree/master/configs/config/pretrain/pirl/models following the screenshot
Screen Shot 2021-02-27 at 1 54 52 PM

3. Make changes to the files in folder https://github.com/facebookresearch/vissl/tree/master/configs/config/pretrain/jigsaw following the screenshot 
Screen Shot 2021-02-27 at 1 54 26 PM

**Comment by akainth015** [comment]
Okay, I have made the changes. you asked for. To be frank though, I am still a bit confused by what result this is expected to produce. The current code, given the following MLP param:
```
[3,4,5,6] Use BN, Use ReLU = False, Skip Last BN ReLU
```
seems as if it would generate the following network
```
Linear(3, 4) -> BN -> Linear (3, 5) -> BN -> Linear(3, 6)
```
because `last_dim` is never updated since ReLU is False.

My understanding is that it should generate the following, hence the reasoning behind my code changes.
```
Linear(3, 3) -> BN -> Linear (3 , 4) -> BN -> Linear(4,5)
```

**Comment by prigoyal** [comment]
> Okay, I have made the changes. you asked for. To be frank though, I am still a bit confused by what result this is expected to produce. The current code, given the following MLP param:
> 
> ```
> [3,4,5,6] Use BN, Use ReLU = False, Skip Last BN ReLU
> ```
> 
> seems as if it would generate the following network
> 
> ```
> Linear(3, 4) -> BN -> Linear (3, 5) -> BN -> Linear(3, 6)
> ```
> 
> because `last_dim` is never updated since ReLU is False.
> 
> My understanding is that it should generate the following, hence the reasoning behind my code changes.
> 
> ```
> Linear(3, 3) -> BN -> Linear (3 , 4) -> BN -> Linear(4,5)
> ```

excellent catch @akainth015 :) We should move the `last_dim = dim` update from line73 out of the `if` conditional and always update the `last_dim`. That should fix. For the example above,

`[3,4,5,6] Use BN=true, Use ReLU = False, Skip Last BN ReLU` -> should give us 

```
Linear(3, 4) -> BN -> Linear (4, 5) -> BN -> Linear(5, 6)
```

**Comment by prigoyal** [comment]
looks great overall. just one last update remaining to `last_dim = dim` moving out of `if` conditional and all good to go otherwise :)

**Review by prigoyal — APPROVED** [review]
looks AMAZING!!! thank you @grace-omotoso and @akainth015 :) you guys rock!!

**Pull request merged**